src/main/java/seedu/contax/ui/TextStyleHelper.java

```diff
@@ -0,0 +1,106 @@
+     */
+    public static List<Text> formattedTextParser(String text) {
+        ArrayList<Text> resultingList = new ArrayList<>();
+        boolean bold = false;
+        boolean italic = false;
+        boolean boldAndItalic = false;
+        boolean monospaced = false;
+        int start = 0;
+
+        for (int i = 0; i < text.length(); i++) {
+            if (text.charAt(i) == '*') {
+                //close bold/italic section
+                if (bold || italic || boldAndItalic) {
+                    if (italic) {
+                        resultingList.add(newText(text.substring(start, i), true, false));
+                        start = i + 1;
+                        italic = false;
+                        continue;
+                    }
+                    if (bold && text.charAt(i + 1) == '*') {
+                        resultingList.add(newText(text.substring(start, i), false, true));
+                        bold = false;
+                        i += 1;
+                        start = i + 1;
+                        continue;
+                    }
+                    if (boldAndItalic && text.charAt(i + 1) == '*' && text.charAt(i + 2) == '*') {
+                        resultingList.add(newText(text.substring(start, i), true, true));
+                        boldAndItalic = false;
+                        i += 2;
+                        start = i + 1;
+                        continue;
+                    }
+                    //Open bold/italic section
+                } else if (text.charAt(i + 1) != '*') {
+                    resultingList.add(newText(text.substring(start, i), false, false));
+                    start = i + 1;
+                    italic = true;
+                } else if (text.charAt(i + 2) != '*') {
+                    resultingList.add(newText(text.substring(start, i), false, false));
+                    start = i + 2;
+                    i += 1;
+                    bold = true;
+                } else {
+                    resultingList.add(newText(text.substring(start, i), false, false));
+                    start = i + 3;
+                    i += 2;
+                    boldAndItalic = true;
+                }
+            } else if (text.charAt(i) == '`') {
+                if (monospaced) {
+                    resultingList.add(newMonospacedText(text.substring(start, i)));
+                    start = i + 1;
+                    monospaced = false;
+                    continue;
+                } else {
+                    resultingList.add(newText(text.substring(start, i), false, false));
+                    start = i + 1;
+                    monospaced = true;
+                }
+            } else if (i == (text.length() - 1)) {
+                resultingList.add(newText(text.substring(start, i + 1), false, false));
+            }
+        }
+        return resultingList;
+    }
+
+    private static Text newText(String text, boolean bold, boolean italic) {
```

Comment:
I suggest renaming this method to something else that does not start with new, perhaps `createTextElement`.

Reply:
thanks for the catch, resolved in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/207/commits/5712c55b1563e8adfdcf92abef182a1197e32964